src/firefly_exchange_client/sockets.py

```diff
@@ -73,6 +74,21 @@ def listener(event, data):
         except:
             pass
         return
+        
+    @sio.event
+    def connect():
+        print("Connected")
+        # add 10 seconds sleep to allow connection to be established before callbacks for connections are executed
+        time.sleep(10)
+        if 'connect' in Sockets.callbacks:
+            # Execute the callback using asyncio.run() if available
```

Comment:
Hey @NidaKarimAli , Was just curious to see how this reconnection would be working. 

Apologies if not able to understand it. Do you feel, you can help me with putting a technical flow (in words) for how it will work?

Reply:
Hey @singhalhemant , thanks for going through the PR. So basically we receive same "connect" event when we connect for the first time or reconnect. For client it's both same events, so if we put a callBack to "connect" event and add all the logic of subscription to that callback, we dont have to worry about doing it manually when connection breaks and auto-reconnects.